contracts/protocols/mainnet/compound_iii/helpers.sol

```diff
@@ -0,0 +1,433 @@
+        assets = new AssetData[](length);
+        AssetInfo memory asset;
+        AssetData memory _asset;
+        Token memory _token;
+        for (uint8 i = 0; i < length; i++) {
+            asset = _comet.getAssetInfo(i);
+
+            TokenInterface token = TokenInterface(asset.asset);
+            _token.offset = asset.offset;
+            _token.token = asset.asset;
+            _token.symbol = token.symbol();
+            _token.decimals = token.decimals();
+            _token.scale = asset.scale;
+
+            _asset.token = _token;
+            _asset.priceFeed = asset.priceFeed;
+            _asset.price = _comet.getPrice(asset.priceFeed);
+            _asset.scale = asset.scale;
+            _asset.borrowCollateralFactor = asset.borrowCollateralFactor;
+            _asset.liquidateCollateralFactor = asset.liquidateCollateralFactor;
+            _asset.liquidationFactor = asset.liquidationFactor;
+            _asset.supplyCap = asset.supplyCap;
+            _asset.totalCollateral = _comet.totalsCollateral(asset.asset).totalSupplyAsset;
+
+            assets[i] = _asset;
+        }
+    }
+
+    function getMarketConfig(address cometMarket) internal view returns (MarketConfig memory market) {
+        IComet _comet = IComet(cometMarket);
+        uint64 utilization = _comet.getUtilization();
+        market.utilization = utilization;
+        market.assetCount = _comet.numAssets();
+        market.supplyRate = _comet.getSupplyRate(utilization);
+        market.borrowRate = _comet.getBorrowRate(utilization);
+        market.baseTrackingSupplySpeed = _comet.baseTrackingSupplySpeed();
+        market.baseTrackingBorrowSpeed = _comet.baseTrackingBorrowSpeed();
+        market.reserves = _comet.getReserves();
+        market.storeFrontPriceFactor = _comet.storeFrontPriceFactor();
+        market.baseBorrowMin = _comet.baseBorrowMin();
+        market.targetReserves = _comet.targetReserves();
+        market.totalSupplyBase = _comet.totalSupply();
+        market.totalBorrowBase = _comet.totalBorrow();
+
+        market.baseToken = getBaseTokenInfo(_comet);
+        market.scales = getScales(_comet);
+        market.rewardConfig = getRewardsConfig(cometMarket);
+        market.assets = getMarketAssets(_comet, market.assetCount);
+    }
+
+    function currentValue(
+        int104 principalValue,
+        uint64 baseSupplyIndex,
+        uint64 baseBorrowIndex,
+        uint64 baseIndexScale
+    ) internal view returns (int104) {
+        if (principalValue >= 0) {
+            return int104((uint104(principalValue) * baseSupplyIndex) / uint64(baseIndexScale));
+        } else {
+            return -int104((uint104(principalValue) * baseBorrowIndex) / uint64(baseIndexScale));
+        }
+    }
+
+    function isAssetIn(uint16 assetsIn, uint8 assetOffset) internal pure returns (bool) {
+        return (assetsIn & (uint16(1) << assetOffset) != 0);
+    }
+
+    function getBorrowableAmount(
```

Comment:
@Richa-iitr Why are you calculating this on the resolver side?

Reply:
I think it can be skipped in the resolver, earlier some params were not returned from the resolver but now I am returning all the details. So should I remove it from the resolver?

Reply:
Since you built it we can keep it for now.

Reply:
@Richa-iitr Maybe you can make this public and remove the connection from getPosition functions

Reply:
@Richa-iitr We can remove healthFactor and sumOfCollateralTimesFactor too

Reply:
Removing all these because it shouldn't cause any minor error because of this and we are not rendering the position on frontend.

Reply:
> @Richa-iitr We can remove healthFactor and sumOfCollateralTimesFactor too

@thrilok209 why?

Reply:
`healthFactor` and `sumOfCollateralTimesFactor` are calculated the same way we do it on the backend. If there is any small error in math calculation, resolver will start to revert and backend APIs won't work.

Reply:
Okay Alright

Reply:
Made the changes